src/test/java/ParserTest.java

```diff
@@ -0,0 +1,80 @@
+import duke.DukeException;
+import duke.Parser;
+import duke.TaskList;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class ParserTest {
+    @Test
+    public void write_ToDoCommand_writtenCorrectly() {
+        Parser testParser = new Parser(new TaskList());
+        String testToDoStr = "todo return book to library";
+
+        try {
+            assertEquals(testParser.parse(testToDoStr), "Got it. I've added this task:\n  " +
+                    "[T][✘] return book to library" + "\nNow you have 1 tasks in the list.");
+            System.out.println("Todo Command correctly parsed.");
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+            fail();
+        }
+    }
+
+    @Test
+    public void write_ToDoCommand_empty_exceptionThrown() {
+        Parser testParser = new Parser(new TaskList());
+        String testToDoStr = "todo ";
+
+        try {
+            assertEquals(testParser.parse(testToDoStr), "Got it. I've added this task:\n  " +
+                    "[T][✘] return book to library" + "\nNow you have 1 tasks in the list.");
+        } catch (DukeException e ) {
+            System.out.println("Exception successfully thrown: " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void write_deadLineCommand_writtenCorrectly() {
+        Parser testParser = new Parser(new TaskList());
+        String testToDoStr = "deadline return book /by 12/12/1212 1212";
+
+        try {
+            assertEquals(testParser.parse(testToDoStr), "Got it. I've added this task:\n  " +
+                    "[D][✘] return book (by: 12/12/1212 1212)" + "\nNow you have 1 tasks in the list.");
+            System.out.println("deadline Command correctly parsed.");
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+            fail();
+        }
+    }
+
+    @Test
+    public void write_deadLineCommand_empty_exceptionThrown() {
+        Parser testParser = new Parser(new TaskList());
+        String testToDoStr = "deadline ";
+
+        try {
+            assertEquals(testParser.parse(testToDoStr), "Got it. I've added this task:\n  " +
+                    "[D][✘] return book (by: 12/12/1212 1212)" + "\nNow you have 1 tasks in the list.");
+            fail();
+        } catch (DukeException e) {
+            System.out.println("Exception successfully thrown: " + e.getMessage());
```

Comment:
I am not to sure about this, but Assert.equals might have to be used to check that the exception is thrown successfully, as printing it out would not allow JUnit to know that the test is passed. 

Also, there is a way to use JUnit to test if methods throw exception.
https://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests
However, I had error when doing this, but you can try it!

Reply:
Hmm, I kind of understood it as if the JUnit tests pass either if 1) fail() is not called or 2) an AssertEquals or any other JUnit-based assertions passes. The printing isn't really there to let JUnit pass the test but rather as a secondary check for the user to see the error message. I suppose I can implement a second AssertEquals against the error message of the exception to see if the expected error message is being thrown. That would be more thorough indeed.

As far as I found, this try-catch method of testing whether an exception is thrown successfully is actually quite standard besides the newer JUnit 5 and 4.7 methods as stated by one of the responders in the link you provided. But it's interesting regardless, I will try out the newer commands. Thank you for the great suggestion!

Reply:
Pardon the double post, but thank you for reviewing the code so thoroughly! It means a lot! (: 💯